engine/src/main/java/org/hibernate/search/engine/logging/impl/Log.java

```diff
@@ -570,4 +571,14 @@ SearchException cannotIncludeAndExcludePathsWithinSameFilter(Set<String> include
 			value = "Multiple configuration providers are available for scope '%1$s'. "
 					+ "They will be taken under consideration in the following order: '%2$s'.")
 	void multipleConfigurationProvidersAvailable(String scope, List<ConfigurationProvider> configurationProviders);
+
+	@Message(id = ID_OFFSET + 121,
+			value = "Trying to use a rewrite method '%1$s' that requires parameter N without specifying the parameter. "
+					+ "Use an overloaded method that allows passing the parameter N.")
```

Comment:
When the user made a mistake, could you please follow this wording? I did a pass around the 6.0 release to make sure we used consistent wording, though I admit things may have drifted a bit since then.

Also I'm afraid "overloaded" may not be newbie-friendly...

```suggestion
			value = "Cannot use rewrite method '%1$s': this method requires parameter 'n', which was not specified."
					+ "Use another version of the rewrite(...) method that accepts parameter 'n'.")
```

Though it would be even better if we used a more self-explanatory name than `n`...

Reply:
for all these "parameterized" rewrites it will eventually reach
https://github.com/apache/lucene/blob/9caeb9395dad6a535882888335915e4ea78e8b43/lucene/core/src/java/org/apache/lucene/search/MultiTermQuery.java#L207

where the parameter is called `size`, but I  thought since we have the `N` in the enum name... it would make sense to keep it as `N`